lib/Provider/future_provider.dart

```diff
@@ -163,18 +164,34 @@ class BrightnessNotifier extends AutoDisposeNotifier<Brightness>
     with WidgetsBindingObserver {
   @override
   void didChangePlatformBrightness() {
-    state = WidgetsBinding.instance.platformDispatcher.platformBrightness;
+    changeBrightness(
+        WidgetsBinding.instance.platformDispatcher.platformBrightness ==
+            Brightness.dark);
     super.didChangePlatformBrightness();
   }
 
   @override
   Brightness build() {
+    bool? isDark = Config().pref.getBool("isDark");
     WidgetsBinding.instance.addObserver(this);
-    return WidgetsBinding.instance.platformDispatcher.platformBrightness;
+    if (isDark == null) {
+      Brightness brightness =
+          WidgetsBinding.instance.platformDispatcher.platformBrightness;
+      brightness == Brightness.dark
+          ? Config().pref.setBool("isDark", true)
+          : Config().pref.setBool("isDark", false);
+      return brightness;
+    } else {
+      return isDark ? Brightness.dark : Brightness.light;
+    }
   }
 
-  void changeBrightness(bool value) =>
-      state = (value) ? Brightness.dark : Brightness.light;
+  Future<void> changeBrightness(bool value) async {
```

Comment:
instead of having one function with bool value being passed in, i think you should make 2 different void functions called setDark and setLight. This will make it easier to use this provider functionality

Reply:
The reason i did not put any async operations is because adding brightness to shared preference has got nothing returning returning the value.
If my operations depended adding values to the shared preference for the first time, then yes it makes sense to "await" the value being put inside the shared preference and then moving forward. However, in this case I don't exactly have to wait for operation to happen since I am not using the result of adding values to shared preference for the rest of the operations.

if bool? isDark = Config().pref.getBool("isDark"); was an asynchronous operation, then yes, I would have to change the Notifier to async notifier since the rest of my operations depended upon the result from the getBool operation. Therefore I will have to "await" for that result of the getBool.

Reply:
I have made that change now please check again @bonamiAkash